unison/taskgroup.go

```diff
@@ -71,11 +71,22 @@ func (c closedGroup) Go(_ func(Canceler) error) error {
 // The zero value of TaskGroup is fully functional. StopOnError must not be set after
 // the first go-routine has been spawned.
 type TaskGroup struct {
-	// StopOnError  configures the behavior when a sub-task failed. If not set
+	// OnQuit  configures the behavior when a sub-task returned. If not set
 	// all other tasks will continue to run. If the function return true, a
 	// shutdown signal is passed, and Go will fail on attempts to start new
 	// tasks.
-	StopOnError func(err error) bool
+	// Next to the action, does the OnQuit error also return the error value
+	// to be recorded. The context.Cancel error will never be recorded.
+	//
+	// Common OnError handlers are given by ContinueOnErrors, StopOnError,
+	// StopOnErrorOrCancel.
+	// By default ContinueOnErrors will be used.
```

Comment:
Why? To me, it sounds counterintuitive that someone would want to continue executing other goroutines in the group if one of them failed. I would add goroutines to the same group if I expected them to stop together when one of them returns an error.

Reply:
The `ContinueOnErrors` handler matches the old semantics. Although it is a breaking change we might want to change the default now (with all the other breaking changes).

> I think StopOnErrorOrCancel is a better default value.

Alternatively I'd rather use `StopOnError` as default. If the group has been cancelled, all workers will get the cancel signal anyways. The `...OrCancel` is more for cases where the cancel signal comes from another path. But the cancel might even be produced by accident e.g. having some code like this (artificial example):

```
tg.Go(func(ctx, context.Context) error {
  ctx, cancel := context.WithCancel(ctx)

  ...

  time.Sleep(5*time.Second)
  cancel()
  return ctx.Err() // here we return an 'inner' context.Cancel, the group is still active.
})
```